Automattic/jetpack
# Cover: Only add filter if needed

Suggestion to fix the `useBlockEditContext` regression for WordPress 5.3.

This emulates the way the VideoPress block is using `useBlockEditContext`, hidden behind an `isSimpleSite` and plan check.

Fixes #16233.

#### Changes proposed in this Pull Request:

* Moves environment checks into `index.js` to only filter the media placeholder when necessary.

#### Does this pull request change what data or activity we track or use?



No

#### Testing instructions:






* Set up a test environment that runs WordPress 5.3.4 with Jetpack master
* Open the editor and try inserting an Image Compare block 
* Make sure the block displays without error

#### Proposed changelog entry for your changes:

* Not needed.
